### DISCUSSION TITLE
[Feature] Function to calculate days until the next holiday

### DISCUSSION THREAD

**Discussion by vlltr** [spatie/holidays#198] [Ideas]
We could consider implementing a function that allows returning the number of days remaining until the next holiday starting from a date passed to the function and the country.

Example:

```php
Holidays::daysUntilNext('sv', '2024-02-01');
// return 52
```

**Comment by Nielsvanpach** [comment]
Carbon already has an option for this. But something like this might be useful:

```php
$next = Holidays::next('sv'); // Returns ['name' => 'New Year', 'date' => CarbonImmutable object]
```

An extra nullable param for a start date might be usefull too.
```php
$next = Holidays::next('sv', startDate: 2024-12-01); // Returns ['name' => 'Christmas', 'date' => CarbonImmutable object]
```

On the Carbon instance you can already count the difference in days.

```php
$days = $next['date']->->diffInDays($date); // Returns int
```
